daisy/disk.go

```diff
@@ -188,6 +192,11 @@ func (dr *diskRegistry) detachHelper(dName, iName string, isAttached bool, s *St
 
 	var att *diskAttachment
 
+	// if the disk has already been attached before workflow is executed, skip validating its attacher
+	if isAttached {
```

Comment:
is this part of the snapshot change or the detach change you did earlier?

Reply:
It's a part of the snapshot change. Snapshot uses a "SourceDisk" which should be validated. If the disk exists before the workflow, we shouldn't verify its attacher.